### PR TITLE
NEW! Username Setup Screen

### PR DESCRIPTION
**Pull request by rauchp**

### CODE REVIEW

**Comment by rauchp** [comment]
Polished and optimized a lot of stuff. Made screens pretty. Tested on Android and iOS.

Simulator Screen Shot - iPhone 11 - 2019-10-21 at 18 41 58


Screenshot_1571704713

File: civilapp/auth/UsernameSetupScreen.js [hunk 1]
```diff
@@ -87,81 +70,123 @@ export default class UsernameSetupScreen extends Component {
...
-    const activeColor = this.state.username ? 'white' : '#454647'
+  changeUsername = (text) => {
+    text = text.toLowerCase()
+    // Localization
+    text = text.normalize('NFD').replace(/[\u0300-\u036f]/g, "")
+    var regex = /[ !@#“‘øßłœ”’$%^&*()_+\-=\[\]{};':"\\|,.<>\/?]/g;
+
+    // Stupid Android Glitch Fix. Typing ab would type aab
+    if (text.length > 1 && text.length - this.state.username.length > 1)
+      text = text.substring(0, text.length - 2) + text[text.length - 1]
```
Comment (EphemeralX) [inline civilapp/auth/UsernameSetupScreen.js]: it's not obvious to me what this is doing; can you add a comment?

File: civilapp/auth/UsernameSetupScreen.js [hunk 2]
```diff
@@ -87,81 +70,123 @@ export default class UsernameSetupScreen extends Component {
...
   }
 
-  render() {
-    const activeLayout = this.state.keyboardVisible ? 'flex-end' : 'center'
-    const activeColor = this.state.username ? 'white' : '#454647'
+  changeUsername = (text) => {
+    text = text.toLowerCase()
+    // Localization
+    text = text.normalize('NFD').replace(/[\u0300-\u036f]/g, "")
+    var regex = /[ !@#“‘øßłœ”’$%^&*()_+\-=\[\]{};':"\\|,.<>\/?]/g;
```
Comment (EphemeralX) [inline civilapp/auth/UsernameSetupScreen.js]: Not entirely sure what we're trying to regex. If we're just trying to remove invalid characters, can't we use [a-zA-Z0-9] or something like that? (common regex expr)

File: civilapp/auth/UsernameSetupScreen.js [hunk 3]
```diff
@@ -87,81 +70,123 @@ export default class UsernameSetupScreen extends Component {
...
+    if (text.length > 1 && text.length - this.state.username.length > 1)
+      text = text.substring(0, text.length - 2) + text[text.length - 1]
+
+    if (!regex.test(text) && text.length < USERNAME_MAX_LENGTH)
+      this.setState({ username: text })
+
+  }
+
+  renderNextButton = (func) => {
+    return <View style={{ margin: 'auto', width: '100%' }} >
```
Comment (EphemeralX) [inline civilapp/auth/UsernameSetupScreen.js]: nit: wrap ()

File: civilapp/auth/UsernameSetupScreen.js [hunk 4]
```diff
@@ -87,81 +70,123 @@ export default class UsernameSetupScreen extends Component {
...
+          autoFocus
+          style={{ height: 0, width: 20 }}
+        />
+        <View style={styles.usernameInputStyle} >
+          <RetroTextInput id="username" value={username} />
         </View>
-      </KeyboardAvoidingView>
+        {showError ? <Text style={styles.enterNameText}>That name already exists</Text> : null}
+        {username.length >= USERNAME_MIN_LENGTH ? this.renderNextButton(() => this.confirmUsername()) : <View style={{ margin: 'auto', width: '100%', height: height / 14 }} />}
+      </View>
```
Comment (EphemeralX) [inline civilapp/auth/UsernameSetupScreen.js]: If this View is just to supply the empty space so it looks right, we only need height right? also can enclose it as <View ... />

File: civilapp/auth/UsernameSetupScreen.js [hunk 5]
```diff
@@ -87,81 +70,123 @@ export default class UsernameSetupScreen extends Component {
...
+    marginTop: 0,
+    // FlexBox
+    flexDirection: 'column',
+    justifyContent: 'center',
+    alignItems: 'center',
+    // Border
+    borderWidth: 4,
+    borderStyle: 'solid',
+    borderRadius: 10,
+    borderColor: '#9BFAFC',
```
Comment (EphemeralX) [inline civilapp/auth/UsernameSetupScreen.js]: not a bad idea

File: civilapp/auth/common/GradientBackground.js [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+        }}
+    </SafeAreaConsumer>
+}
+const styles = {
+    center: {
+        flex: 1,
+        justifyContent: 'center',
+        alignItems: 'center'
+    },
+
```
Comment (EphemeralX) [inline civilapp/auth/common/GradientBackground.js]: n

File: civilapp/auth/common/GradientBackground.js [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+const gradientColors = [
+    '#64B8A4',
+    '#279F82',
+    '#69F3D2',
+    '#32BD9B',
+    '#399C84',
+]
+
+export default GradientBackground = (props) => {
```
Comment (EphemeralX) [inline civilapp/auth/common/GradientBackground.js]: not a bad idea.

File: civilapp/auth/common/SuperText.js [hunk 8]
```diff
@@ -1,86 +1,132 @@
...
+        <Text style={[styles.numberTextInputText, { color: 'black', zIndex: -1, fontSize: size, paddingRight: 7, paddingTop: 7 }]}>
+            {props.value}
+        </Text>
+        <Text style={[styles.numberTextInputText, { color: 'black', zIndex: -1, fontSize: size, paddingRight: 7 }]}>
+            {props.value}
+        </Text>
+        <Text style={[styles.numberTextInputText, { color: 'black', zIndex: -1, fontSize: size, paddingLeft: 7 }]}>
+            {props.value}
+        </Text>
+
```
Comment (EphemeralX) [inline civilapp/auth/common/SuperText.js]: https://media1.tenor.com/images/30b8e61e6caf0de6751058c967718f0d/tenor.gif?itemid=15111424

File: civilapp/android/.settings/org.eclipse.buildship.core.prefs [hunk 9]
```diff
@@ -0,0 +1,2 @@
+connection.project.dir=
+eclipse.preferences.version=1
```
[the remark this answers is not in the document]
Comment (rauchp) [reply civilapp/android/.settings/org.eclipse.buildship.core.prefs]: oWo I think a grade update installed it

File: civilapp/android/app/src/main/AndroidManifest.xml [hunk 10]
```diff
@@ -42,7 +42,7 @@
         android:name=".MainActivity"
         android:label="@string/app_name"
         android:configChanges="keyboard|keyboardHidden|orientation|screenSize"
-        android:windowSoftInputMode="adjustResize"
+        android:windowSoftInputMode="adjustNothing"
```
[the remark this answers is not in the document]
Comment (rauchp) [reply civilapp/android/app/src/main/AndroidManifest.xml]: before views would automatically shrink to accommodate keyboard, we can get the same effect with KeyboardAvoidingView, so by doing this we standardize iOS and Android, and avoid unintentionally shrinking view which is what was happening on Android.

File: civilapp/auth/SignInScreen.js [hunk 11]
```diff
@@ -221,29 +207,32 @@ export default class SignInScreen extends Component {
     return <View style={{ margin: 'auto', width: '100%' }} >
       <View style={{ alignItems: 'center' }}>
         <TouchableOpacity disabled={this.state.showLoading} onPress={func}>
-          <ImageLoader style={{ height: height / 14, resizeMode: 'contain' }} source={!this.state.showLoading ? Nextbtn : NextbtnLoad} />
+          <ImageLoader style={{ height: width * .20, width: width * .75, resizeMode: 'cover' }} source={!this.state.showLoading ? Nextbtn : NextbtnLoad} />
```
[the remark this answers is not in the document]
Comment (rauchp) [reply civilapp/auth/SignInScreen.js]: Yeah, aspect ratio tings

File: civilapp/auth/common/SuperText.js (hunk 8 above)
Comment (rauchp) [reply civilapp/auth/common/SuperText.js]: forgive me coding jesus for what I am going to push to prod

**Pull request merged**